dask_ml/model_selection/_successive_halving.py

```diff
@@ -211,7 +218,16 @@ def __init__(
             tol=tol,
             random_state=random_state,
             scoring=scoring,
+            verbose=verbose,
         )
+        self.n_initial_parameters = n_initial_parameters
+        self.n_initial_iter = n_initial_iter
+        self.aggressiveness = aggressiveness
+
+        # unusual ordering allows BaseIncrementalSearchCV to have a prefix
```

Comment:
What does "unusual ordering" mean here?

Reply:
Explained in 9df6914d0d920c19fe9bb9925c3c0a2943098533

Reply:
In general, subclasses should implement a superset of their parent's API: https://en.wikipedia.org/wiki/Liskov_substitution_principle

So if the only reason for this is `prefix` not really being necessary for IncrementalSearchCV, I'd prefer to just add it.

Reply:
My motivation more clearly: I don't want to expose `prefix` in a public API (like in `IncrementalSearchCV.__init__`). I think it's an internal detail; Scikit-learn doesn't expose it in [RandomizedSearchCV]. I'd prefer not to an attribute after the fact (`search._prefix = ...`).

Maybe the best solution is to create a new dummy class `_IncrementalSearchCV` to pass `prefix` to `BaseIncrementalSearchCV`. Then, `IncrementalSearchCV` and `SuccessiveHalvingSearchCV` could inherit from the new class.

[RandomizedSearchCV]:https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.RandomizedSearchCV.html

Reply:
I don't have a strong opinion on whether or not `prefix` is in the public API, but I do care that all of these have compatible signature. So if we need `prefix` in the base class so that we can consistently do logging, then we need to have it in all of the signatures.

Reply:
I've satisfied both our concerns with a private class to pass `prefix` to `BaseIncrementalSearchCV`:

* `prefix` is not in the public API
* All subclasses implement a superset of the parent's API.

Briefly, the implementation looks like this:

``` python
class BaseIncrementalSearchCV(...):
    ...
class _IncrementalSearchCV(BaseIncrementalSearchCV):
    ...
class IncrementalSearchCV(_IncrementalSearchCV):
    ...
class SuccessiveHalvingSearchCV(_IncrementalSearchCV):
    ...
```

Reply:
I'll look more closely, but I'm not a fan of the extra indirection. And I think `BaseIncrementalSearchCV` has the parameter, and so all the subclasses (including IncrementalSearchCV) need it.

It also seems like it might be a generally useful parameter? If I'm running multiple IncrementalSearchCV's simultaneous I might want to provide a prefix to monitor their performance?

Reply:
> might want to provide a prefix to monitor their performance?

I suppose in the context of Dask it does make sense – that's the motivation for `prefix` in `SuccessiveHalvingSearchCV` too, and I can see how it could apply to my current hyperparameters optimization task.

I'll add `prefix` to the public API for `IncrementalSearchCV`.

Reply:
...but the algorithm-specific parameter `decay_rate` is accepted by `IncrementalSearchCV`. `decay_rate` is not used by any other adaptive algorithm, including `SuccessiveHalvingSearchCV`.

I think that's work for #432, not this PR.